tests/unit/hook_packages/terraform/hooks/prepare/test_resource_linking.py

```diff
@@ -131,3 +132,41 @@ def test_resource_full_address_root_module(self):
         module = TFModule(None, None, {}, {}, {}, {})
         resource = TFResource("resource_address", "type", module, {})
         self.assertEqual(resource.full_address, "resource_address")
+
+    @patch("samcli.hook_packages.terraform.hooks.prepare.resource_linking._resolve_module_variable")
+    @patch("samcli.hook_packages.terraform.hooks.prepare.resource_linking._get_configuration_address")
+    def test_resolve_module_output_with_var(self, config_mock, resolve_var_mock):
+        module = TFModule("", None, {"mycoolref": "mycoolvar"}, [], {}, {"mycooloutput": References(["var.mycoolref"])})
+
+        config_mock.return_value = "mycoolref"
+
+        _resolve_module_output(module, "mycooloutput")
+
+        config_mock.assert_called_with("mycoolref")
+        resolve_var_mock.assert_called_with(module, "mycoolref")
```

Comment:
can we assert that we return what we received from `_resolve_module_variable`

Reply:
At the moment, this method isn't implemented (won't actually return anything). Additionally, should we be testing the output of a different method in this test, since this test is for outputs?

Reply:
My comment was not clear, sorry for that. What I meant is to let this mock function return some value, and let's your unit test validate that what `_resolve_module_output` return is what `_resolve_module_variable` mock function return. In your testing you validate that we call `_resolve_module_variable` with the correct value, but you do not validate how `_resolve_module_output` will use `_resolve_module_variable` output.